public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -270,10 +270,9 @@ export class ElasticDatasource
     const annotation = options.annotation;
     const timeField = annotation.timeField || '@timestamp';
     const timeEndField = annotation.timeEndField || null;
-    const dashboard = options.dashboard;
 
     // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-    const adhocVariables = dashboard.getVariables().filter((v) => v.type === 'adhoc') as AdHocVariableModel[];
+    const adhocVariables = this.templateSrv.getVariables().filter((v) => v.type === 'adhoc') as AdHocVariableModel[];
```

Comment:
this is not how you get adhoc filters, call getAdHocFilters()

Reply:
ideally they should get them from the request object but looks like with the legacy annotationsQuery function we do not pass ad-hoc filters